### PR TITLE
corefile uploader: Updates per review comments offline

### PR DESCRIPTION
**Pull request by renukamanavalan**
1) core_uploader service waits for syslog.service
2) core_uploader service enabled for restart on failure
3) Use mtime instead of file size + ample time to be robust.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by renukamanavalan** [comment]
I have not tested the code yet. Can I get your comments please, before I do my last round testing.

**Comment by renukamanavalan** [comment]
Changes:
1) The service dependency switched to syslog.service 
2) file_write_complete switch to watching st_mtime instead of size.
3) Added some time buffer to ensure all write completes.
4) Upon failure to upload, keep retrying every hour with an error message,

File: files/image_config/corefile_uploader/core_uploader.py [hunk 1]
```diff
@@ -221,18 +229,18 @@ def handle_file(path):
...
 
     @staticmethod
-    def upload_file(fname, fpath):
+    def upload_file(fname, fpath, coref):
         daemonname = fname.split(".")[0]
         i = 0
         fail_msg = ""
         
-        while i <= MAX_RETRIES:
+        while True:
```
Comment (lguohan) [inline files/image_config/corefile_uploader/core_uploader.py]: I think you can do retry in the big loop. if it fails, retry in the big loop.

File: files/image_config/corefile_uploader/core_uploader.service [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-Requires=updategraph.service
-After=updategraph.service
+Requires=syslog.service
+After=syslog.service
 
 [Service]
 Type=simple
 ExecStart=/usr/bin/core_uploader.py
+StandardOutput=null
+Restart=on-failure
```
Comment (lguohan) [inline files/image_config/corefile_uploader/core_uploader.service]: we need back off here? if core_uploader is constantly restarting?

Comment (renukamanavalan) [reply files/image_config/corefile_uploader/core_uploader.service]: No need. The only failure that crashes the service, is some fatal things like, sonic_version.yaml does not have expected attributes or some fatal system related failure (which should crash many more ...), and running out of disk space.

The one thing, I would need to take care of "running out of disk space". This I can take care of inside the script.

File: files/image_config/corefile_uploader/core_uploader.py (hunk 1 above)
Comment (renukamanavalan) [reply files/image_config/corefile_uploader/core_uploader.py]: I guess no. The bigger loop is either scan/wait-for-core. In either case, the next one would suffer the same fate. So I rather spew log & retry, until either I succeed or service is restarted, by someone alerted by these log messages.

**Comment by renukamanavalan** [comment]
retest vs please.

**Comment by renukamanavalan** [comment]
vs & vsimage build failure are unrelated to this PR.

console o/p:
"
_```
Step 6/28 : RUN apt-get update   &&     apt-get install -y          curl                    ca-certificates         gcc                     make                    libdpkg-perl            ipmitool
 ---> Running in bf1b21cffd19
Ign:1 http://debian-archive.trafficmanager.net/debian stretch InRelease
Hit:2 http://debian-archive.trafficmanager.net/debian-security stretch/updates InRelease
Hit:3 http://debian-archive.trafficmanager.net/debian stretch-backports InRelease
Hit:4 http://debian-archive.trafficmanager.net/debian stretch Release
Reading package lists...
[91mE: Release file for http://debian-archive.trafficmanager.net/debian/dists/stretch-backports/InRelease is expired (invalid since 12min 13s). Updates for this repository will not be applied.
[0mThe command '/bin/sh -c apt-get update   &&     apt-get install -y          curl                    ca-certificates         gcc                     make                    libdpkg-perl            ipmitool' returned a non-zero code: 100
[  FAIL LOG END  ] [ target/docker-snmp-sv2.gz ]
slave.mk:526: recipe for target 'target/docker-snmp-sv2.gz' failed
```_
"

**Pull request merged**